### PR TITLE
Add check_distribution function

### LINKED ISSUE #4170: Implement `check_distribution` function for pipelines

### PR DESCRIPTION
**Pull request by eccabay**
Closes #4170

### CODE REVIEW

File: evalml/tests/model_understanding_tests/test_metrics.py [hunk 1]
```diff
@@ -631,3 +632,22 @@ def test_jupyter_graph_check(
...
+
+@pytest.mark.parametrize(
+    "problem_type",
+    ["binary", "multiclass", "regression", "time series regression"],
+)
+@pytest.mark.parametrize("expected_distribution", ["matching", "not matching"])
+def test_check_distribution(problem_type, expected_distribution):
+    y_true = pd.Series(np.random.normal(size=500))
+    if expected_distribution == "matching":
+        y_pred = pd.Series(np.random.normal(size=500))
```
Comment (jeremyliweishih) [inline evalml/tests/model_understanding_tests/test_metrics.py:645]: would we want to test non-normal distributions?

File: evalml/tests/model_understanding_tests/test_metrics.py [hunk 2]
```diff
@@ -631,3 +632,22 @@ def test_jupyter_graph_check(
         graph_roc_curve(y, y_pred_proba)
         assert len(graph_valid) == 0
         import_check.assert_called_with("ipywidgets", warning=True)
+
+
+@pytest.mark.parametrize(
+    "problem_type",
+    ["binary", "multiclass", "regression", "time series regression"],
+)
+@pytest.mark.parametrize("expected_distribution", ["matching", "not matching"])
```
Comment (jeremyliweishih) [inline evalml/tests/model_understanding_tests/test_metrics.py:641]: think we need a test case for the baseline case where only one class is predicted

**Review by jeremyliweishih — APPROVED** [review]
LGTM once we add the baseline case for coverage

File: evalml/tests/model_understanding_tests/test_metrics.py (hunk 2 above)
Comment (eccabay) [reply evalml/tests/model_understanding_tests/test_metrics.py:641]: That's what the `non_matching` case checks!

**Review by chukarsten — APPROVED** [review]
Looks great! Well done.

**Pull request merged**